packages/veritone-widgets/src/redux/modules/mediaDetails/index.js

```diff
@@ -0,0 +1,657 @@
+  ) {
+    const errorMessage = get(error, 'message', error);
+
+    let previousResultsByEngineId =
+      state[widgetId].engineResultsByEngineId || {};
+    let engineResultRequestsById =
+      state[widgetId].engineResultRequestsByEngineId;
+    // It is possible results were requested by
+    const resultsGroupedByEngineId = groupBy(payload, 'engineId');
+    forEach(resultsGroupedByEngineId, (results, engineId) => {
+      if (!previousResultsByEngineId[engineId]) {
+        // Data hasn't been retrieved for this engineId yet
+        previousResultsByEngineId[engineId] = map(results, 'jsondata');
+      } else {
+        // New results need to be merged with previously fetched results
+        let insertionIndex = findIndex(previousResultsByEngineId[engineId], {
+          startOffsetMs,
+          stopOffsetMs,
+          status: 'FETCHING'
+        });
+
+        // TODO: fitler out any duplicate data that overflows time chunks.
+        previousResultsByEngineId[engineId] = [
+          ...previousResultsByEngineId[engineId].slice(0, insertionIndex),
+          ...map(results, 'jsondata'),
+          ...previousResultsByEngineId[engineId].slice(insertionIndex + 1)
+        ];
+      }
+
+      engineResultRequestsById[engineId] = engineResultRequestsById[
+        engineId
+      ].map(request => {
+        if (
+          request.startOffsetMs === startOffsetMs &&
+          request.stopOffsetMs == stopOffsetMs &&
+          request.status === 'FETCHING'
+        ) {
+          return {
+            ...request,
+            status: 'SUCCESS'
+          };
+        }
+        return request;
+      });
+    });
+
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: !error || null,
+        error: error ? errorMessage : null,
+        engineResultsByEngineId: {
+          ...previousResultsByEngineId
+        },
+        engineResultRequestsByEngineId: {
+          ...engineResultRequestsById
+        }
+      }
+    };
+  },
+  [LOAD_TDO](state, { meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: null,
+        error: null,
+        warning: null,
+        tdo: null
+      }
+    };
+  },
+  [LOAD_TDO_SUCCESS](state, { payload, meta: { warn, error, widgetId } }) {
+    const tdo = payload;
+    const errorMessage = get(error, 'message', error);
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: !(warn || error) || null,
+        error: error ? errorMessage : null,
+        warning: warn || null,
+        tdo: tdo
+      }
+    };
+  },
+  [UPDATE_TDO](state, { meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: null,
+        error: null,
+        warning: null
+      }
+    };
+  },
+  [UPDATE_TDO_COMPLETE](state, { payload, meta: { warn, error, widgetId } }) {
```

Comment:
see my comment for `LOAD_ENGINE_CATEGORIES_COMPLETE`

Reply:
done